### PR TITLE
feat: wishlist functionality added

### PR DESCRIPTION
**Pull request by rahulyadav139**
Demo: https://palasio-lane-store.netlify.app/

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,25 +1,27 @@
...
 
 function App() {
   const { isAuthModal, resetModal, switchModal, isAuthTypeLogin } =
     useAuthModal();
 
+  const { isAuth } = useAuth();
+
   return (
     <div className="App">
       <Routes>
```
Comment (bharati-21) [inline src/App.js:22]: Suggestion (you can work on this later) - But you can move this to a separate file and them import the component here/

File: src/components/authorization/LoginForm.js [hunk 2]
```diff
@@ -56,17 +55,21 @@ const LoginForm = props => {
       return console.log('invalid password');
```
Comment (bharati-21) [inline src/components/authorization/LoginForm.js:55]: Please remove logs.

File: src/components/authorization/LoginForm.js [hunk 3]
```diff
@@ -56,17 +55,21 @@ const LoginForm = props => {
...
     }
 
+    console.log('login');
```
Comment (bharati-21) [inline src/components/authorization/LoginForm.js:58]: here as well.

File: src/components/authorization/SignupForm.js [hunk 4]
```diff
@@ -1,9 +1,9 @@
 import './AuthForm.css';
-import { useInput } from '../../hooks/use-input';
+import { useInput, useAuth, useAuthModal } from '../../hooks';
 import { textFormatter } from '../../utils';
 import { useState, useEffect, Fragment } from 'react';
-import { useAuth } from '../../contexts/auth-context';
-import { useAuthModal } from '../../contexts/auth-modal-context';
+// import { useAuth } from '../../contexts/auth-context';
+// import { useAuthModal } from '../../contexts/auth-modal-context';
```
Comment (bharati-21) [inline src/components/authorization/SignupForm.js:6]: Please remove commented out code.

File: src/components/listing-page/filters/CarModels.js [hunk 5]
```diff
@@ -10,22 +10,17 @@ const CarModels = props => {
...
-     
+
       return;
     }
 
     const newArr = props.carModels.filter(el => el !== e.target.value);
-  
+
     props.onGetCarModels(newArr);
-
```
Comment (bharati-21) [inline src/components/listing-page/filters/CarModels.js:28]: Suggestion - you could destructure props and then use the prop items passed.

File: src/components/listing-page/product-card/ProductCard.js [hunk 6]
```diff
@@ -11,36 +15,52 @@ const ProductCard = props => {
...
+        >
+          {isAuth && wishlist.items.includes(prodId) ? (
+            <i class="fas fa-heart"></i>
+          ) : (
+            <i className="far fa-heart"></i>
+          )}
         </button>
       </div>
-      <Link to={`/product/${prodId}`}>
-        <h2 className="product-title">{title.substring(0, 22) + '...'}</h2>
```
Comment (bharati-21) [inline src/components/listing-page/product-card/ProductCard.js:42]: Suggestion - you could do this with CSS (you can look into line-clamp and text-overflow: elipsis; later)

File: src/components/single-product-page/SingleProductCard.js [hunk 7]
```diff
@@ -1,12 +1,54 @@
...
+
+    wishlist.items.includes(prodId)
+      ? removeFromWishlist(prodId)
+      : addToWishlist(prodId);
+  };
+
+  const wishlistButton =
+    isAuth && wishlist.items.includes(prodId) ? (
+      <button
+        onClick={toggleWishListHandler.bind(null, prodId)}
```
Comment (bharati-21) [inline src/components/single-product-page/SingleProductCard.js:32]: I'm not sure why bind is being used here. any explanations?

File: src/hooks/use-context.js [hunk 8]
```diff
@@ -0,0 +1,10 @@
+import { useContext } from 'react';
+import { AuthContext, AuthModalContext, WishlistContext } from '../contexts';
+
+const useAuth = () => useContext(AuthContext);
+
+const useAuthModal = () => useContext(AuthModalContext);
+
+const useWishlist = () => useContext(WishlistContext);
+
+export { useAuth, useAuthModal, useWishlist };
```
Comment (bharati-21) [inline src/hooks/use-context.js:10]: Nice!

File: src/hooks/use-fetch.js [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+        headers,
+        body: JSON.stringify(body),
+      });
+
+      status = res.status;
+
+      data = await res.json();
+    } catch (err) {
+      error = err;
+      console.log(err);
```
Comment (bharati-21) [inline src/hooks/use-fetch.js:27]: Please remove logs

File: src/pages/cart/Cart.js [hunk 10]
```diff
@@ -4,7 +4,7 @@ import {
   PriceBreakoutCard,
   Header,
   Footer,
-} from '../components';
+} from '../../components';
 import { Fragment } from 'react';
 // import { CartProductCard } from '../components/cart/CartProductCard';
```
Comment (bharati-21) [inline src/pages/cart/Cart.js:9]: Please remove commented code.

File: src/components/listing-page/product-card/ProductCard.js (hunk 6 above)
Comment (rahulyadav139) [reply src/components/listing-page/product-card/ProductCard.js:42]: Sure, thank you for pointing that out.

File: src/pages/product-detail-page/ProductDetailPage.js [hunk 11]
```diff
@@ -4,25 +4,27 @@ import {
...
-      } catch (err) {
-        console.log(err);
-      }
+      const { data, error } = await getData(
+        `http://localhost:8080/product/${prodId}`,
+        false
+      );
+
+      if (error) return;
+      setProduct(data);
```
Comment (bharati-21) [inline src/pages/product-detail-page/ProductDetailPage.js:27]: Why not handle it with catch()?

File: src/pages/product-listing-page/ProductListingPage.js [hunk 12]
```diff
@@ -87,4 +117,4 @@ const ProductListingPage = props => {
     </Fragment>
   );
 };
```
Comment (bharati-21) [inline src/pages/product-listing-page/ProductListingPage.js:119]: Please remove commented out code.

**Review by bharati-21 — COMMENTED** [review]
Please remove logs and commneted code.

Apart from that, everything looks good!

File: src/components/single-product-page/SingleProductCard.js (hunk 7 above)
Comment (rahulyadav139) [reply src/components/single-product-page/SingleProductCard.js:32]: I am using it here for binding the argument. Here there no "this" keyword available so I am passing null as the first argument.

File: src/pages/product-detail-page/ProductDetailPage.js (hunk 11 above)
Comment (rahulyadav139) [reply src/pages/product-detail-page/ProductDetailPage.js:27]: I am handling it in a custom hook (useFetch)

File: src/pages/product-listing-page/ProductListingPage.js [hunk 13]
```diff
@@ -12,33 +12,63 @@ const ProductListingPage = props => {
...
+  //   (async () => {
+  //     console.log('app page');
+  //     const { error } = await sendData(
+  //       'http://localhost:8080/admin/wishlist',
+  //       'PUT',
+  //       wishlist,
+  //       true
+  //     );
+  //   })();
+  // }, [quantity]);
```
Comment (parulgupta2001) [inline src/pages/product-listing-page/ProductListingPage.js:33]: remove commented code

File: src/pages/product-listing-page/ProductListingPage.js [hunk 14]
```diff
@@ -12,33 +12,63 @@ const ProductListingPage = props => {
...
+    //     const res = await fetch(
+    //       `http://localhost:8080/products?filterBy=${filterBy}`
+    //     );
+    //     const data = await res.json();
+
+    //     setProducts(data);
+    //     setLoading(false);
+    //   } catch (err) {
+    //     console.log(err);
+    //   }
```
Comment (parulgupta2001) [inline src/pages/product-listing-page/ProductListingPage.js:58]: here also remove commented code

**Review by parulgupta2001 — APPROVED** [review]
code look good

**Pull request merged**